python/ray/_private/worker.py

```diff
@@ -1000,6 +1000,7 @@ def init(
     log_to_driver: bool = True,
     namespace: Optional[str] = None,
     runtime_env: Optional[Union[Dict[str, Any], "RuntimeEnv"]] = None,  # noqa: F821
+    metadata: List[Tuple[str, str]] = None,
```

Comment:
We probably want to implement the underlying functionality first, before making the option available in the public API. Also,
- `metadata` sounds a bit too generic. How about `request_headers`?
- We need a way to pass this to `ray start` command, in `scripts.py`

Reply:
@mwtian Can we create an additional issue for appending this to the ray start command? This has to do with the SDK and gRPC client specifically, so uncertain why it needs to be added with ray start.

Reply:
I see. `ray.init()` is a public API. We probably need an approval to add a parameter.

Since this is going to be Ray client specific, how about we use an environment variable like `RAY_CLIENT_GRPC_HEADER` to pass the information?

cc @scv119 @pcmoritz @ericl @zhe-thoughts